mne/simulation/evoked.py

```diff
@@ -4,16 +4,59 @@
 #
 # License: BSD (3-clause)
 import copy
+import warnings
 
 import numpy as np
 
 from ..io.pick import pick_channels_cov
 from ..forward import apply_forward
-from ..utils import check_random_state, verbose, _time_mask
+from ..utils import check_random_state, verbose, _time_mask, deprecated
+
+
+@deprecated('"generate_evoked" is deprecated and will be removed in '
+            'MNE-0.11. Please use simulate_evoked instead')
+def generate_evoked(fwd, stc, evoked, cov, snr=3, tmin=None,
+                    tmax=None, iir_filter=None, random_state=None,
+                    verbose=None):
+    """Generate noisy evoked data
+
+    Parameters
+    ----------
+    fwd : dict
+        a forward solution.
+    stc : SourceEstimate object
+        The source time courses.
+    evoked : None | Evoked object
+        An instance of evoked used as template.
+    cov : Covariance object
```

Comment:
Sometimes it is Covariance object and sometimes it's instance of Covariance (same for evoked...), which one is to use?